### PR TITLE
[WIP] First attempt at adding LocalAddr to etcd client v3

### PR DESCRIPTION
**Pull request by doekerqhtv**
This PR will allow etcd administrator to select outgoing IP address bind in systems with multiple NIC cards, or IP addresses or aliases.

### CODE REVIEW

File: etcdmain/config.go [hunk 1]
```diff
@@ -131,6 +131,7 @@ func newConfig() *config {
 	fs.Var(cfg.ec.CorsInfo, "cors", "Comma-separated white list of origins for CORS (cross-origin resource sharing).")
 	fs.StringVar(&cfg.ec.Dir, "data-dir", cfg.ec.Dir, "Path to the data directory.")
 	fs.StringVar(&cfg.ec.WalDir, "wal-dir", cfg.ec.WalDir, "Path to the dedicated wal directory.")
+	fs.StringVar(&cfg.ec.LocalAddr, "local-address", cfg.ec.LocalAddr, "Local adddress bind for outgoing client connections.")
```
Comment (xiang90) [inline etcdmain/config.go]: we need to mention this is for peer communication as other flags. probably also add a peer prefix.

**Comment by xiang90** [comment]
do you have a test plan for this new flag?

File: etcdmain/config.go (hunk 1 above)
Comment (doekerqhtv) [reply etcdmain/config.go]: What do you suggest? peer-local-address?

**Comment by fanminshi** [comment]
@doekerqhtv add a comment to explain the purpose of this pr?

File: etcdmain/config.go (hunk 1 above)
Comment (xiang90) [reply etcdmain/config.go]: yea.

**Pull request closed without merging**